### DISCUSSION TITLE
It would be nice to have a list of preconditions for testing of PR's

### DISCUSSION THREAD

**Discussion by treblereel** [Vertispan/j2clmavenplugin#164] [General]
To speed up our development we should have a strict check list of preconditions:

compilation modes we support:

- BUNDLE
- BUNDLE_JAR
- ADVANCED

reactor mode:
- single
- multimodule project

OS:
- linux
- osx
- windows

java versions:
- 11
- 17

what else ?

**Comment by niloc132** [comment]
build/test vs watch

multi-module: test when running watch from parent or from child

include java 8 when testing versions (note that CI already tests java 8 and 11, 5 different maven builds, all three OSes)
